frontend/public/components/dashboard/inventory-card/inventory-item.tsx

```diff
@@ -45,102 +54,161 @@ const getStatusGroupIcons = (flags: FlagsObject) => {
 };
 
 export const InventoryItem: React.FC<InventoryItemProps> = React.memo(
-  ({ isLoading, singularTitle, pluralTitle, count, children, error = false }) => {
-    const title = count !== 1 ? pluralTitle : singularTitle;
-    let status: React.ReactNode;
-    if (error) {
-      status = <div className="co-dashboard-text--small text-secondary">Unavailable</div>;
-    } else if (isLoading) {
-      status = <LoadingInline />;
-    } else {
-      status = children;
-    }
-    return (
-      <div className="co-inventory-card__item">
-        <div className="co-inventory-card__item-title">{isLoading || error ? title : `${count} ${title}`}</div>
-        <div className="co-inventory-card__item-status">{status}</div>
+  ({ isLoading, title, count, children, error = false, TitleComponent, ExpandedComponent }) => {
+    const [expanded, setExpanded] = React.useState(false);
+    const pluralizedTitle = count !== 1 ? `${title}s` : title;
+    const titleMessage = isLoading || error ? pluralizedTitle : `${count} ${pluralizedTitle}`;
+    return ExpandedComponent ? (
+      <Accordion asDefinitionList={false} headingLevel="h5" className="co-inventory-card__accordion">
+        <AccordionItem>
+          <AccordionToggle
+            onClick={() => setExpanded(!expanded)}
```

Comment:
If we really wanted to use hooks consistently and thoroughly, we'd use `React.useCallback` here :smiley: and everywhere else where we pass callbacks as props in JSX.

That said, I'm OK with anonymous functions. They're created upon each render and thus (AFAIK) trigger VDOM diffing but it shouldn't be a big deal unless the components are too heavy (complex DOM structure).

Reply:
since I like `React.useCallback`, I opted for using it, thanks! :)